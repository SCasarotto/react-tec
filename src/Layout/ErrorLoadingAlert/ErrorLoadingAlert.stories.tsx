import React from 'react';

import { Story, Meta } from '@storybook/react';

import { ErrorLoadingAlert, ErrorLoadingAlertProps } from './ErrorLoadingAlert';

export default {
  title: 'Layout/ErrorLoadingAlert',
  component: ErrorLoadingAlert,
  argTypes: {},
} as Meta;

const Template: Story<ErrorLoadingAlertProps> = (args) => (
  <ErrorLoadingAlert {...args} />
);

export const Default = Template.bind({});
Default.args = {
  title: 'Error Loading',
  message: 'There appears to be an error loading the data you requested.',
  onClick: () => alert('Button Clicked'),
  buttonTitle: 'Go Back',
};
