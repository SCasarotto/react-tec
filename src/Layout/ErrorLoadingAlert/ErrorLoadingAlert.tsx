import React from 'react';

import {
  PanelWrapper,
  Panel,
  Title,
  Message,
  Button,
} from './styledComponents';

export interface ErrorLoadingAlertProps {
  className?: string;
  title?: React.ReactNode;
  message?: React.ReactNode;
  buttonTitle?: string;
  onClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
}
/**
 *
 * ErrorLoadingAlert in a simple component built inside of a `PanelWrapper` and `Panel` typically used when there is a loading error for a whole page.
 *
 */
export const ErrorLoadingAlert: React.FC<ErrorLoadingAlertProps> = (props) => {
  const {
    className = '',
    title,
    message,
    onClick,
    buttonTitle,
    children,
  } = props;
  return (
    <PanelWrapper className={`tec-errorLoadingAlert ${className}`}>
      <Panel>
        {title && <Title className='tec-errorLoadingAlertTitle'>{title}</Title>}
        {message && (
          <Message className='tec-errorLoadingAlertMessage'>{message}</Message>
        )}
        {buttonTitle && (
          <Button onClick={onClick} className='tec-errorLoadingAlertButton'>
            {buttonTitle}
          </Button>
        )}
        {children}
      </Panel>
    </PanelWrapper>
  );
};
