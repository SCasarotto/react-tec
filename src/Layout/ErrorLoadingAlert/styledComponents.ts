import styled from 'styled-components';

import { Button as B } from '../../Form/Button';
import { Panel as P } from '../Panel';
import { PanelWrapper as PW } from '../PanelWrapper';

export const PanelWrapper = styled(PW)`
  justify-content: center;
`;
export const Panel = styled(P)`
  max-width: 600px;
  margin: 30px auto;
`;
export const Title = styled.h1`
  margin-top: 0px;
  margin-bottom: 20px;
  text-align: center;
  font-size: 26px;
`;
export const Message = styled.p`
  margin-top: 0px;
  margin-bottom: 20px;
  text-align: center;
  font-size: 16px;
`;
export const Button = styled(B)`
  margin: 10px auto;
  width: auto;
`;
