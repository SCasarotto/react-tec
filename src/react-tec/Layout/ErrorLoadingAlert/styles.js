//import { colors, layout } from './../../../config/styles';

export default {
	container : {
		justifyContent : 'center',
	},
	errorLoadingContainer : {
		maxWidth    : 600,
	},
	errorTitle: {
		marginTop       : 0,
		marginBottom    : 20,
		textAlign       : 'center',
		fontSize        : 26
	},
	errorMessage: {
		marginTop 		: 0,
		marginBottom 	: 20,
		textAlign 		: 'center',
	},
}
