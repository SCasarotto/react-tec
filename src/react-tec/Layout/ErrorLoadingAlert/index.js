import ErrorLoadingAlert from './ErrorLoadingAlert';

export default ErrorLoadingAlert;
