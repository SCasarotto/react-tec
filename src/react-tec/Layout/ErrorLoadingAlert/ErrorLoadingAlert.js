//
//TE Version 0.2.0
//

import React, { Component } from 'react'
import Radium from 'radium'

import PanelWrapper from './../PanelWrapper'
import Panel from './../Panel'
import TEButton from './../../Form/TEButton'
import styles from './styles'

class ErrorLoadingAlert extends Component {
	render() {
		const { title, message } = this.props
		return (
			<PanelWrapper style={styles.container}>
				<Panel containerStyle={styles.errorLoadingContainer}>
					<h1 style={styles.errorTitle}>{title}</h1>
					<p style={styles.errorMessage}>{message}</p>
				</Panel>
			</PanelWrapper>
		)
	}
}

ErrorLoadingAlert.propTypes = {}

ErrorLoadingAlert.defaultProps = {}

export default Radium(ErrorLoadingAlert)
