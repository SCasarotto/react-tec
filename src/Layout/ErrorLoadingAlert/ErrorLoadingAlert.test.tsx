import React from 'react';

import { render } from '@testing-library/react';

import { ErrorLoadingAlert } from './ErrorLoadingAlert';

describe('ErrorLoadingAlert', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<ErrorLoadingAlert />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(<ErrorLoadingAlert className='myClassName' />);
    expect(container.firstChild).toHaveClass('myClassName');
  });
});
