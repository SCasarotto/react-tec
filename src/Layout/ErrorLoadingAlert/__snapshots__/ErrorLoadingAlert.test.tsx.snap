// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorLoadingAlert matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  position: relative;
  border-radius: 5px;
  padding: 30px;
  margin-bottom: 30px;
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 30px;
}

.c1 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  max-width: 600px;
  margin: 30px auto;
}

@media (max-width:800px) {
  .c2 {
    padding: 25px;
    margin-bottom: 25px;
  }
}

@media (max-width:650px) {
  .c2 {
    padding: 20px;
    margin-bottom: 20px;
  }
}

@media (max-width:450px) {
  .c2 {
    padding: 15px;
    margin-bottom: 15px;
    border-radius: 3px;
  }
}

@media (max-width:800px) {
  .c0 {
    padding: 25px;
  }
}

@media (max-width:650px) {
  .c0 {
    padding: 20px;
  }
}

@media (max-width:450px) {
  .c0 {
    padding: 15px;
  }
}

<div
    class="c0 tec-panelWrapper c1 tec-errorLoadingAlert "
  >
    <div
      class="c2 tec-panel c3"
    />
  </div>
</DocumentFragment>
`;
